jupyterlab/labapp.py

```diff
@@ -214,16 +224,19 @@ def _validate(self, data, base_url, page_url, workspaces_url):
         if 'data' not in workspace:
             raise Exception('The `data` field is missing.')
 
-        if 'metadata' not in workspace:
-            raise Exception('The `metadata` field is missing.')
+        # See if workspace name is given, inject into workspace.
+        if self.workspace_name != '':
```

Comment:
I think we also want another branch here for if the `--name` argument is given as the empty string. In that case the workspace id would just be that of the default JupyterLab workspace (i.e. `/lab`). I'm not sure the best way to accomplish that via the CLI, though. Can we distinguish between `None` and `""` using this traitlets-based approach?

Reply:
> Can we distinguish between None and "" using this traitlets-based approach?

I don't think so. I think `None` is an invalid type for a `Unicode` trait.

Reply:
We can catch an empty string though and turn it into the default workspace.

Reply:
Actually, I just found out that Traits have a keyword argument `allow_none`. This should solve the problem. Updating the PR now.

Reply:
I'm don't that this will fix the entire problem. The UI I am thinking of is something like
1. If no `--name` arg is given, then the importer uses the same ID that it currently has.
2. If a `--name="foo"` is given, the workspace ID becomes `/user/username/lab/foo`.
3. If a `--name=""` is given, the workspace ID becomes that of the default workspace, i.e. `/user/username/lab`.

I don't think that the current behavior will hit case number 3. That is to say, we need special logic for the default workspace URL since it is different from all the named workspace URLs.

I don't have too strong of a grasp of how the traitlets command-line processing happens -- is a distinction made between `--name=""` vs `--name=`?

Reply:
> is a distinction made between `--name=""` vs `--name=`?

No. traitlets assumes the empty space after `--name=` is an empty string. To get `None`, the user has to explicitly pass `--name=None` *or* not use `--name`.